metaflow/plugins/__init__.py

```diff
@@ -20,38 +45,40 @@ def get_plugin_cli():
 from .aws.batch.batch_decorator import BatchDecorator, ResourcesDecorator
 from .conda.conda_step_decorator import CondaStepDecorator
 
-STEP_DECORATORS = [CatchDecorator,
+STEP_DECORATORS = _merge_lists([CatchDecorator,
                    TimeoutDecorator,
                    EnvironmentDecorator,
                    ResourcesDecorator,
                    RetryDecorator,
                    BatchDecorator,
-                   CondaStepDecorator]
+                   CondaStepDecorator], ext_plugins.STEP_DECORATORS, 'name')
 
 # Add Conda environment
 from .conda.conda_environment import CondaEnvironment
-ENVIRONMENTS = [CondaEnvironment]
+ENVIRONMENTS = _merge_lists([CondaEnvironment], ext_plugins.ENVIRONMENTS, 'TYPE')
 
 
 # Every entry in this list becomes a class-level flow decorator.
 # Add an entry here if you need a new flow-level annotation. Be
 # careful with the choice of name though - they become top-level
 # imports from the metaflow package.
 from .conda.conda_flow_decorator import CondaFlowDecorator
-FLOW_DECORATORS = [CondaFlowDecorator]
+FLOW_DECORATORS = _merge_lists([CondaFlowDecorator], ext_plugins.FLOW_DECORATORS, 'name')
 
 # Sidecars
-SIDECAR = {}
+SIDECAR = ext_plugins.SIDECAR
```

Comment:
Nit: This should've ideally been `SIDECARS` since its a dictionary (so a container of a sidecars).
If you agree - feel free to handle this in a separate PR. But just want to field for your opinion via this comment.

Reply:
Done.